### DISCUSSION TITLE
[Feature] Sort people

### DISCUSSION THREAD

**Discussion by JesseWebDotCom** [immich-app/immich#4879] [Feature Request]
### The feature

Right now people are sorted by the most matches - which makes it hard to find the face you are looking for. How about additional sort options like alphabetical?

### Platform

- [ ] Server
- [ ] Web
- [ ] Mobile

**Comment by upsetdog** [comment]
Most matches, Least matches, A-Z ascending, A-Z descending (not sure how that works in languages like Japanese though). Maybe date of birth too?

**Reply by celsius6** [reply]
When managing a large photo library with hundreds of recognized people, the current People view becomes increasingly difficult to navigate.

Users often need to:

Find a specific person among many unnamed entries.
Review recently detected people.
Prioritize people with the largest number of associated photos.
Organize and clean up face recognition results efficiently.

Currently, the People view does not provide sorting options, which makes large-scale face management cumbersome.

Proposed Solution

Add configurable sorting options to the People page.

Suggested sorting modes:

Alphabetical
Name (A → Z)
Name (Z → A)

Useful when many people have already been named.

Asset Count
Most photos first
Fewest photos first

Useful for prioritizing the most important people and cleaning up duplicates.

Recently Added
Newest people first
Oldest people first

Useful after importing new photos or running facial recognition on a large library.

Recently Updated
Recently merged
Recently renamed
Recently modified

Useful during ongoing face-recognition maintenance.

Additional Ideas
Allow users to choose a default sort order.
Persist the selected sort order across sessions.
Combine sorting with existing filtering options.
Display the number of associated assets directly in the People overview.
Benefits
Easier management of large libraries.
Faster cleanup of facial recognition results.
Improved user experience for users with hundreds or thousands of detected people.
Better support for long-term maintenance of People data.

Thank you for considering this feature.

**Comment by lucianomunoz** [comment]
yes please, its so dumb

**Comment by sagar5534** [comment]
Perhaps keep the current implementation and create "favorite" faces. This way your favourited faces will show up first

**Reply by bigstusexy** [reply]
I don't believe this really addresses the issue.  

Let's say you're thinking of a person and you know you have them tagged.  You think their name is either Betty, Sally, or Mulva . Especially on mobile where you cannot search by name, you're going to have to search the list of people, possibly three full times.  If you have hundreds of names, this is very time consuming.  

If this were in Alphabetical order, you could at least cut it down to just searching all of B, C, and M beginning names.  Being able to set the criteria helps cut it down in other ways.  Like you know you meet them only once before and probably have less than 5 photos of them (though I don't think count is listed)

**Reply by arnolicious** [reply]
I'm currently implementing this in #14866 FYI.

This is only the favorite function, favorited people will be shown first on the web version.

**Comment by bizzarrone** [comment]
I would like to sort people too, expecially on mobile app where no filed to search is present.
And more, when show/hide people hided, after some item, the watermark of slashed eye disappear.

And could be really helpful to modify the flag "hidden" into a photo where that people appears

**Reply by bgeneto** [reply]
I'm finding it extremely challenging to locate specific individuals' names within the mobile app, especially when dealing with thousands of faces.

 Unfortunately, the face search feature has become nearly unusable for me. It would be greatly appreciated if a keyword search function could be implemented in the mobile app, allowing users to search for specific names using a format such as: 

person:"John Doe". 

This would significantly improve the overall user experience.

**Comment by Dezc** [comment]
I'd like to sort people too ; by Name, by number of picture, number of albums!

**Comment by diyoyo** [comment]
I recently upgraded to 1.135.1, and I realize People are now sorted alphabetically, even the favorite ones 

Is there an option to revert back to the old style?

PS: I'm referring to the web version here.

**Reply by bo0tzz** [reply]
This is currently being polled and discussed on the discord server.

**Reply by mareg** [reply]
Upvoting for either allowing us to sort People, or reverting it to sorting by number of matches 👍🏻

**Reply by marvin78** [reply]
Sorting by number of matches is logical. It should be sortable by the user but if that's difficult it should be by count.

**Reply by trop1kal** [reply]
Upvoting also! I liked the previous way of sorting by number of matches.

**Comment by bo0tzz** [comment]
After reverting to sorting by asset count, that also establishes we will not be implementing this feature request.

**Reply by dima-ser** [reply]
Please reconsider adding sort by name as an option. Sorting my number of matches can be default, but sorting by name would really help finding duplicate names.

**Comment by dima-ser** [comment]
It would really be useful to have the ability to sort by name. I have external libraries with existing faces and Immich often splits the same person into multiple people for some reason (even though they have exact same name). I've been dealing with it by merging them, but trying to find duplicate people in a grid of a thousand is nearly impossible. Having the ability to sort by name would make this trivial.

**Reply by bigstusexy** [reply]
I didn't have a discord account (don't want one either) and never got there.  What I wondered is why is this not a switch?  Something people can choose. 

I understand that people like different things.  To me, there is no value in listing people by how many tagged photos I have of them, I have no need for a popularity list, but others like it.  The search / filter is great but if I'm looking for a person and start with what I think the first letter of their name is Alphabetical order will help.  Also if I get use to their name being close to other names.

**Comment by ChilledSlim** [comment]
Sorting by match count frequency just makes sense.  A bunch of **people in the background of photos** (match count 3-5) are well ahead of my actual friends (match count 90) and mixed in with family (match count ~1000).

**Reply by dima-ser** [reply]
> Sorting by match count frequency just makes sense. A bunch of **people in the background of photos** (match count 3-5) are well ahead of my actual friends (match count 90) and mixed in with family (match count ~1000).

It makes sense for some cases, but not others like mine. Why not provide an option so users can choose however they want to sort?

**Comment by pqne** [comment]
I would like to propose a "weighted frequency" sorting option. Maybe it solves a problem only I have, but to me it makes more sense than sorting by frequency (which I agree is the next best option).

My situation: I'm in my early 30s, live in a big city, and my social circle is not as stable as it probably will be once I'm 50 or 60. Friends come into my life, other friends move away from the city. I have a partner for some years, then we break up; we still talk but I don't see them much anymore. 
I don't want to ban or hide any of those people from my life/immich; but neither does it make much sense to have my ex, who I have thousands of photos of but who I haven't seen in 8 years, show up 2nd on my people list.

The idea: sort faces by frequency, but include a "weight" factor that decreases from 1 for today to let's say 0.1 for 5 years ago. Like this, a person who I have 100 photos of from this year gets sorted ahead of a person who I have 1000 photos of, but from 6 years ago. 

I understand this has several drawbacks: 
- it's hard to transparently explain to users without getting mathematical 
- it requires some experimenting to figure out what is a good formula and parameters (linear vs. exponential decrease, decay rate, ...)
- maybe my problem is universal and resonates with you, but it might just be different enough to not work out: either your life changes faster than mine, or it is way more stable, which could mean different parameters
- ... or you prefer simple frequency and just hide long lost loved ones; or there's a better way (other than changing my "favourite people" every year)? Please let me know!

I also understand this might only make sense to implement once (if ever) a decision is made to give the user a choice of sorting options (occurrence, alphabetically, weighted, ...). Still, to me this seems like a very sensible approach to sorting. I'm curious to hear your thoughts.

**Comment by lucianomunoz** [comment]
@pqne i love that idea as an other alternative of sorting, for sure!.

to me the biggest problem is why there is a resistance to give us options, i understand have an option to be by what ammount of pictures, but let me order it by name also or by country or whatever :)

**Comment by lucianomunoz** [comment]
👏🏻👏🏻👏🏻👏🏻

love all these options, would be great!


Luciano A. Muñoz Sessarego
Web: www.pintamonos.cl
Email: ***@***.***
Vimeo: www.vimeo.com/lucianomunoz
Gumroad: https://gumroad.com/lucianomunoz
Imdb: https://www.imdb.com/name/nm8355386/
Linkedin: http://www.linkedin.com/in/lucianomunoz
